test/compilable/test16085.d

```diff
@@ -0,0 +1,13 @@
+// REQUIRED_ARGS: -de
+// PERMUTE_ARGS:
+import imports.imp16085;
+
+void test()
```

Comment:
Why not make this a runnable test and assert that the correct overload is called? You can make each member of overload set return a different value and test the return value. I know that this PR is about visubility and not overloading, but it would make the test more clear and useful.